File: src/elisp/modes/sample.js
/**
 * Complex JavaScript sample for highlighting
 */
import { EventEmitter } from 'events';

const GLOBAL_CONSTANT = 42;

class ThemeValidator extends EventEmitter {
  constructor(name) {
    super();
    this.name = name;
    this.regex = /^[a-z0-9-]+$/i;
  }

  /**
   * Validates a recipe object
   */
  async validate(recipe) {
    try {
      if (!recipe.id || typeof recipe.id !== 'string') {
        throw new Error('Invalid ID');
      }

      console.log(`Validating ${this.name}...`);
      
      const isValid = this.regex.test(recipe.id);
      this.emit('validation', { isValid, timestamp: Date.now() });

      return isValid;
    } catch (err) {
      console.error(`Error: ${err.message}`);
      return false;
    }
  }
}

const validator = new ThemeValidator('EmacsThemes');
validator.on('validation', (result) => {
  const { isValid } = result;
  console.log(`Result: ${isValid ? 'PASSED' : 'FAILED'}`);
});

const sampleRecipe = {
  id: 'zenburn-theme',
  tags: ['dark', 'low-contrast'],
  complexity: GLOBAL_CONSTANT
};

validator.validate(sampleRecipe);

File: src/elisp/modes/sample.cpp
#include <iostream>
#include <vector>
#include <string>
#include <memory>
#include <algorithm>

/**
 * Complex C++ sample for highlighting
 */

namespace ThemeSystem {

class Theme {
public:
    virtual ~Theme() = default;
    virtual void apply() const = 0;
    virtual std::string getName() const = 0;
};

class DarkTheme : public Theme {
private:
    std::string name;
    std::vector<std::string> tags;

public:
    DarkTheme(std::string n) : name(std::move(n)) {
        tags = {"dark", "high-contrast"};
    }

    void apply() const override {
        std::cout << "Applying dark theme: " << name << std::endl;
        for (const auto& tag : tags) {
            std::cout << "  Tag: " << tag << std::endl;
        }
    }

    std::string getName() const override { return name; }
};

} // namespace ThemeSystem

int main() {
    using namespace ThemeSystem;
    
    auto themes = std::vector<std::unique_ptr<Theme>>();
    themes.push_back(std::make_unique<DarkTheme>("Zenburn"));
    themes.push_back(std::make_unique<DarkTheme>("Monokai"));

    std::for_each(themes.begin(), themes.end(), [](const auto& t) {
        t->apply();
    });

    return 0;
}